### ISSUE TITLE
Outlook categories - include or exclude from sync

### ISSUE THREAD

**Issue by workjoeea** [derekantrican/GAS-ICS-Sync#313]
Is there a way to include or exclude certain categories from an Outlook calendar from syncing?

**Comment by jonas0b1011001** [comment]
Until we have a more sophisticated filtering option implemented you can manually add filters in Helpers.gs [L184](https://github.com/derekantrican/GAS-ICS-Sync/blob/e0924b9ea4ca657f02b42588674efbe1b2a7f80f/Helpers.gs#L184)

```
result = result.filter(function(event){
    try{
      return (event.getFirstPropertyValue('categories').toString() != "Category to exclude");
    }catch(e){
      return true;
    }
  });
```

**Comment by workjoeea** [comment]
thank you @jonas0b1011001 ...this is excellent.  I'm not against doing things manually, but I'm a little confused as to execution.  I would have to add this string for every category that I'd like to exclude?  That's okay, I just need to know the process.  Thank you.

**Comment by jonas0b1011001** [comment]
If you need to exclude multiple categories:
`return !(["exclude 1","exclude 2","exclude 3"].includes(event.getFirstPropertyValue('categories').toString()));`

or to only sync specific caategories:
`return (["include 1","include 2","include 3"].includes(event.getFirstPropertyValue('categories').toString()));`

**Comment by workjoeea** [comment]
@jonas0b1011001 thank you for that clarification.

**Comment by workjoeea** [comment]
Ok, I gave it a try, tells me syntax error.  So I clearly misunderstand something.  Here's my line 184-190:

result = result.filter(function(event){
    try{
      return !(["Blue category","Yellow category","Green category","Orange category","Purple category","Red category"].includes(event.getFirstPropertyValue('categories').toString() != "Category to exclude");
    }catch(e){
      return true;
    }
  });

Thank you for your help so far.

**Comment by jonas0b1011001** [comment]
Should be
```
result = result.filter(function(event){
    try{
      return !(["Blue category","Yellow category","Green category","Orange category","Purple category","Red category"].includes(event.getFirstPropertyValue('categories').toString()));
    }catch(e){
      return true;
    }
  });
```

**Comment by workjoeea** [comment]
Ok, so I copied and pasted exactly this code to line 184 in helper.gs, and yet the excluded categories still sync.  Thoughts?

**Comment by jonas0b1011001** [comment]
Download your *.ics, open it in a texteditor and check if outlook is actually using the _categories_ parameter, might be some nonstandard outlook parameter. If so, change the _getFirstPropertyValue('categories')_ accordingly.

**Comment by workjoeea** [comment]
Thanks for that tip.  I don't see "Categories" anywhere, nevermind a parameter that shows any of the colors (I just searched the text).  Is there a way for me to make Outlook generate that in the ICS file?  Or is there another solution?

**Comment by jonas0b1011001** [comment]
I don't know much about outlook, maybe @derekantrican has an idea.

**Comment by cephas20012** [comment]
Hi, I'm trying to follow along too. My work calendar is too detailed to share with my family, I want to just share the Personal and Info Share categories. I've saved my Outlook ics and verify it is using categories, for example one of my events with this line:
BEGIN:VEVENT
CATEGORIES:Info Share

I tried both ways, listing the include filters with the return ( line or the exclude filters with the return !( line but still all of my calendar is showing up.

There is a block of code for cancelled events from L176-183 already with a result = result.filter(function(event){

I tried just adding my block below at L184, or do I need to edit the cancelled events result.filter?

```
  result = result.filter(function(event){
    try{
      return (["Personal","Info Share"].includes(event.getFirstPropertyValue('CATEGORIES').toString()));
    }catch(e){
      return true;
    }
  });
```
